lib/getContractSources.js

```diff
@@ -5,21 +5,54 @@ import path from 'path'
  * Extract contents of contract files as string
  *
  * @param  {Array}  contracts             Contracts to compile
+ * @param  {String} modulesPath           Path to node_modules directory
  * @param  {String} inputPath             Directory to read target contracts from
  * @param  {String} contractFileExtension File extension of the target contracts
  * @return {Array}                        Promises resolving contract file content
  */
-const getContractSources = (contracts, inputPath, contractFileExtension) => {
+const getContractSources = (
+  contracts,
+  modulesPath,
+  inputPath,
+  contractFileExtension
+) => {
   return contracts.map(async contract => {
     try {
-      // read contract file content
-      const input = await fs.readFileSync(
-        path.resolve(`${inputPath}/${contract}.${contractFileExtension}`),
-        'utf8'
+      // Check for the file in inputPath directory
+      const inputPathExists = await fs.pathExists(
+        `${inputPath}/${contract}.${contractFileExtension}`
       )
 
-      // return key/value of contractName.contractFileExtension and file content
-      return { [`${contract}.${contractFileExtension}`]: input }
+      if (inputPathExists) {
+        // Read contract file content
+        const input = fs.readFileSync(
+          path.resolve(`${inputPath}/${contract}.${contractFileExtension}`),
+          'utf8'
+        )
+
+        // Return key/value of contractName.contractFileExtension and file content
+        return { [`${contract}.${contractFileExtension}`]: input }
+      }
+
+      // File could not be found in CWD, check in user module directories
+      return new Promise(resolve =>
+        module.paths.map(async modulePath => {
```

Comment:
I think mapping over all `module.paths` could cause issues:
a) If the module is found in the first path, there is no need to search the others
b) It could be possible that the module is found in more than one path (eg. local and global) and would then be compiled twice.

So, perhaps either a `for` loop with a `break` or `Array.some`?